libraries/standard/mqtt/src/mqtt.c

```diff
@@ -997,6 +1000,19 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
         status = MQTT_DeserializeAck( pIncomingPacket, NULL, pSessionPresent );
     }
 
+    /* If a clean session is requested, a session present should not be set by
+     * broker. */
+    if( status == MQTTSuccess )
+    {
+        if( ( cleanSession == true ) && ( *pSessionPresent == true ) )
+        {
+            LogError( ( "Received CONNACK with session present from broker."
+                        " This should not happen as a clean session is requested"
+                        " in CONNECT." ) );
```

Comment:
Probably should be a warning instead of an error as this represents an issue in the broker,
Also, recommend making the message concise as:
```suggestion
            LogWarn( ( "Unexpected session present flag in CONNACK response from broker."
                        " CONNECT request with clean session was made with broker." ) );
```

Reply:
I think I still would prefer the error as it is a condition in which a connection cannot be continued with. I like the sentences you suggested :)